django/contrib/postgres/aggregates/general.py

```diff
@@ -8,20 +11,46 @@
 ]
 
 
-class ArrayAgg(OrderableAggMixin, Aggregate):
+# RemovedInDjango50Warning
+class _DeprecatedConvertValueMixin:
+    _not_provided = object()
+
+    def __init__(self, *expressions, default=_not_provided, **extra):
+        if default is self._not_provided:
+            default = None
+            self._default_provided = False
+        else:
+            self._default_provided = True
+        super().__init__(*expressions, default=default, **extra)
+
+    def convert_value(self, value, expression, connection):
+        if value is None and not self._default_provided:
+            warnings.warn(
+                self._deprecation_message,
+                category=RemovedInDjango50Warning,
+            )
+            return self._deprecation_value
+        return value
+
+
+class ArrayAgg(_DeprecatedConvertValueMixin, OrderableAggMixin, Aggregate):
     function = 'ARRAY_AGG'
     template = '%(function)s(%(distinct)s%(expressions)s %(ordering)s)'
     allow_distinct = True
 
+    # RemovedInDjango50Warning
+    _deprecation_value = property(lambda self: [])
+    _deprecation_message = (
+        'From Django 5.0, ArrayAgg() will return None instead of an empty '
```

Comment:
"From Django 5.0" isn't clear to me. Does it mean, "Starting with Django 5.0"? If so, some other ways to say this could be "Starting in Django 5.0," "In Django 5.0 and up," "From Django 5.0 onward," or perhaps simply, "In Django 5.0" if the behavior won't carry forward.

Reply:
"In Django 5.0" :+1:

Reply:
I'll go with "In Django 5.0, ..." as we've done something similar before:

https://github.com/django/django/blob/37ec8721df493413d5ea5266b6faead9b8f520a9/django/test/testcases.py#L1071-L1077